### PR TITLE
add validation methods and properties

### PR DESCRIPTION
**Pull request by ch1234**

### CODE REVIEW

File: stripe-elements.html [hunk 1]
```diff
@@ -477,6 +524,31 @@
       }
     }
 
+    /**
+     * Checks if the Stripe form is valid.
+     * @return {Boolean} true if the Stripe form is valid
+     */
+    validate() {
+      let isValid = this.isComplete && !this.isEmpty && !this.hasError;
```
Comment (bennypowers) [inline stripe-elements.html]: Prefer `const` over `let`

File: stripe-elements.html [hunk 2]
```diff
@@ -477,6 +524,31 @@
...
+      if (!isValid) {
+        let error = {
```
Comment (bennypowers) [inline stripe-elements.html]: Same here, and prefer concise object notation i.e.

```js
const message = "...";
this._setError({message});
```

File: stripe-elements.html [hunk 3]
```diff
@@ -477,6 +524,31 @@
...
+      return isValid;
+    }
+
+    /**
+     * Checks for potential validity. A potentially valid form is one that
+     * is not empty, not complete and has no error. A validated form also counts
+     * as potentially valid.
+     * @return {Boolean} true if the Stripe form is potentially valid
+     */
+    isPotentiallyValid() {
```
Comment (bennypowers) [inline stripe-elements.html]: Interesting... What's the use case?

Comment (ch1234) [reply stripe-elements.html]: It's something Square offers on their payment form. For me personally, I won't use it but it was easy to add in while I put in the rest of the methods so I thought it didn't hurt to add some additional functionality. I guess a dev could show some styling once the form isn't empty and is potentially a valid input.

**Comment by bennypowers** [comment]
Good going thanks a mint!

image of super baby goat

**Pull request merged**